File: Node/src/controllers/accounts.controller.js
module.exports.register = async (req, res) => {
    res.send(await req.service.register(req));
}

module.exports.login = async (req, res) => {
    res.send(await req.service.login(req, res));
};

module.exports.logout = async (req, res) => {
    res.send(await req.service.logout(req, res));
};

module.exports.getNewToken = async (req, res) => {
    res.send(await req.service.getNewToken(req, res));
}

module.exports.updatePassword = async (req, res) => {
    res.send(await req.service.updatePassword(req));
}

module.exports.deleteAccount = async (req, res) => {
    res.send(await req.service.deleteAccount(req));
}

File: dotNET/WebAPI/Controllers/AccountsController.cs
using Bussiness.Accounts.Commands.DeleteAccount;
using Bussiness.Accounts.Commands.GetNewToken;
using Bussiness.Accounts.Commands.Logout;
using Bussiness.Accounts.Commands.Register;
using Bussiness.Accounts.Commands.UpdatePassword;
using Bussiness.Accounts.Dtos;
using Bussiness.Accounts.Queries.Login;
using Core.Utilities.Results;
using MediatR;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace WebAPI.Controllers
{
    public class AccountsController : BaseController
    {
        public AccountsController(IMediator mediator) : base(mediator)
        { }

        [HttpPost("login")]
        public async Task<Result<LoginInfo>> Login(LoginQuery loginQuery)
        {
            return await Mediator.Send(loginQuery);
        }

        [HttpPost("logout")]
        [Authorize]
        public async Task<Result> Logout()
        {
            return await Mediator.Send(new LogoutCommand());
        }

        [HttpPost("register")]
        public async Task<Result> Register(RegisterCommand registerCommand)
        {
            return await Mediator.Send(registerCommand);
        }

        [Authorize]
        [HttpPut]
        public async Task<Result> UpdatePassword(UpdatePasswordCommand updatePasswordCommand)
        {
            return await Mediator.Send(updatePasswordCommand);
        }

        [Authorize]
        [HttpDelete]
        public async Task<Result> DeleteAccount()
        {
            return await Mediator.Send(new DeleteAccountCommand());
        }

        [HttpPost("token")]
        public async Task<Result> GetNewAccessToken()
        {
            return await Mediator.Send(new GetNewTokenCommand());
        }
    }
}
